File: Problem2.cpp
#include<iostream>
#include<stdio.h>
using namespace std;

int main(){
  long i = 0, j = 1, sum = 0, temp;
  while(j < 4000000){
    temp = i; i = j; j = temp + j;
    if(j % 2 == 0)
      sum += j;
  }
  cout << sum;
}

File: java/Problem2.java
public class Problem2 {
    public static void main(String[] args){
        // Self explanatory. Standard fibonacci gen
        long a = 0, b = 1, temp;
        long sum = 0;
        while(a < 4000000){
            if (b % 2 == 0) sum += b;
            temp = a;
            a = b;
            b = temp + b;
        }
        System.out.println(sum);
    }
}
